src/Helper/NumberHelper.php

```diff
@@ -0,0 +1,348 @@
+    {
+        $formatter = $this->getFormatter($locale ?? $this->localeDetector->getLocale(), \NumberFormatter::CURRENCY, $attributes, $textAttributes, $symbols);
+        $number = $this->parseNumericValue($number);
+
+        return $this->fixCharset($formatter->formatCurrency($number, $currency));
+    }
+
+    /**
+     * Formats a number in scientific notation according to the specified
+     * locale and \NumberFormatter attributes.
+     *
+     * @param string|float|int $number         The number to format
+     * @param array            $attributes     The attributes used by \NumberFormatter
+     * @param array            $textAttributes The text attributes used by \NumberFormatter
+     * @param array            $symbols        The symbols used by \NumberFormatter
+     * @param string|null      $locale         The locale used to format the number
+     *
+     * @return string The formatted number
+     */
+    public function formatScientific($number, array $attributes = [], array $textAttributes = [], array $symbols = [], ?string $locale = null): string
+    {
+        $number = $this->parseNumericValue($number);
+
+        return $this->format($number, \NumberFormatter::SCIENTIFIC, $attributes, $textAttributes, $symbols, $locale);
+    }
+
+    /**
+     * Formats a number as ordinal according to the specified locale and
+     * \NumberFormatter attributes.
+     *
+     * @param string|float|int $number         The number to format
+     * @param array            $attributes     The attributes used by \NumberFormatter
+     * @param array            $textAttributes The text attributes used by \NumberFormatter
+     * @param array            $symbols        The symbols used by \NumberFormatter
+     * @param string|null      $locale         The locale used to format the number
+     *
+     * @return string The formatted number
+     */
+    public function formatOrdinal($number, array $attributes = [], array $textAttributes = [], array $symbols = [], ?string $locale = null): string
+    {
+        $number = $this->parseNumericValue($number);
+
+        return $this->format($number, \NumberFormatter::ORDINAL, $attributes, $textAttributes, $symbols, $locale);
+    }
+
+    /**
+     * Formats a number according to the specified locale and \NumberFormatter
+     * attributes.
+     *
+     * @param string|float|int $number         The number to format
+     * @param int              $style          The Style used by the formatter
+     * @param array            $attributes     The attributes used by the formatter
+     * @param array            $textAttributes The text attributes used by the formatter
+     * @param array            $symbols        The symbols used by \NumberFormatter
+     * @param string|null      $locale         The locale used to format the number
+     */
+    public function format($number, int $style, array $attributes = [], array $textAttributes = [], array $symbols = [], ?string $locale = null): string
```

Comment:
do we need to make all format methods public? or the class should be used with the other methods? is this used outside this class?